src/dls_bluesky_core/plans/scanspec.py

```diff
@@ -0,0 +1,79 @@
+import operator
+from functools import reduce
+from typing import Any, List, Mapping, Optional
+
+import bluesky.plans as bp
+from bluesky.protocols import Movable, Readable
+from cycler import Cycler, cycler
+from scanspec.specs import Spec
+
+from dls_bluesky_core.core import MsgGenerator
+
+"""
+Plans related to the use of the `ScanSpec https://github.com/dls-controls/scanspec`
+library for constructing arbitrarily complex N-dimensional trajectories, similar to
+Diamond's "mapping scans" using ScanPointGenerator.
+"""
+
+
+def scan(
+    detectors: List[Readable],
+    axes_to_move: Mapping[str, Movable],
```

Comment:
Would quite like this to be dealt with before/during the move. This plan should be a simple, base scan plan usable like this

```python
def my_plan():
    yield from scan([my_detector], Line(x, 0, 10, 10))
```

Reply:
Did you make any progress on this?

Reply:
Nope, bring it in as-is, we'll have to break it later, I have negative bandwidth at this point